### ISSUE TITLE
select first result as I type

### ISSUE THREAD

**Issue by jeffgca** [mozilla/universal-search#266]
This is a muscle memory problem from the Awesomebar. As I start typing the first result should be selected so I can just hit Enter instead of down-arrow then Enter.

**Comment by chuckharmston** [comment]
Yeah, 100% agree that this is less than ideal. Some project history:

We begun by trying to replace the entire awesomebar popup with an [`iframe`](https://github.com/mozilla/universal-search-content) and [communication channel](https://github.com/mozilla/universal-search-addon) allowing the `iframe` to receive the same data that the existing XUL-based architecture does.

It worked and gave us a great deal of control over the experience, but we had trouble getting it to parity with XUL in terms of performance and responsiveness. That made the reliability of the data we wanted to collection questionable (i.e. high attrition rates could be because it was slow, rather than because users didn't appreciate the content). So, in order to best answer the core question—do users engage with server-provided recommendations?—we decided to shelve the `iframe` in favor of a closer-to-native approach that inserts a row on top of the existing popup.

Pro: it's very responsive and feels quite fast.
Con: we are much more restricted in how and when we can tweak the UX.

It turns out (and @6a68 can speak to the specifics better than I), one of those restrictions relates to highlight management and when/how we're able to steal the highlight from and pass it to the search results. A chain of bugs and pull requests that discuss this, if you're interested in the dirty details:
- https://github.com/mozilla/universal-search/pull/80
- https://github.com/mozilla/universal-search/pull/116
- https://github.com/mozilla/universal-search/issues/118

So tl;dr: we agree, and it's not great. It's a decision consciously made for gnarly technical reasons that we will likely invest time into fixing if we decide to continue pursuing the concept.

Thanks for the file!
